official/update.rkt

```diff
@@ -3,31 +3,94 @@
          racket/function
          racket/system
          racket/package
+         racket/port
          racket/match
+         racket/async-channel
          pkg/private/stage
          (prefix-in pkg: pkg/lib)
          "common.rkt"
          "notify.rkt"
          "static.rkt")
 
+(define NUM-THREADS 4)
```

Comment:
How did you choose 4?

Reply:
I can run a proper experiment to see what a good number is. In my initial run (which re-fetches the entire database), I used 8 threads, but it only gives 4x speedup. That's why I settled with only 4 threads here. It's also a number that doesn't seem to be too high.

Reply:
Iirc The reason I didn't use threads in the past was because the EC2 instance couldn't handle it. I seem to remember it was slower. Maybe I am misremembering or things have changed, but make sure you test on the actual server and not just locally

Reply:
On an AWS instance provided by @samdphillips (which has the same environment as the actual server)

1 thread: 72 mins
2 threads: 49 mins
4 threads: 42 mins
8 threads: 41 mins